modules/javafx.controls/src/main/java/javafx/scene/chart/XYChart.java

```diff
@@ -1316,11 +1371,37 @@ protected void invalidated() {
                 Node node = get();
                 if (node != null) {
                     node.accessibleTextProperty().unbind();
+                    ObservableValue<String> seriesLabel = seriesProperty
+                            .flatMap(Series::nameProperty)
+                            .orElse("");
+                    ObservableValue<String> xAxisLabel= seriesProperty
+                            .flatMap(Series::chartProperty)
+                            .flatMap(XYChart::xAxisProperty)
+                            .flatMap(Axis::labelProperty)
+                            .orElse(ControlResources.getString("XYChart.series.xaxis"));
+                    ObservableValue<String> yAxisLabel = seriesProperty
+                            .flatMap(Series::chartProperty)
+                            .flatMap(XYChart::yAxisProperty)
+                            .flatMap(Axis::labelProperty)
+                            .orElse(ControlResources.getString("XYChart.series.yaxis"));
                     node.accessibleTextProperty().bind(new StringBinding() {
-                        {bind(currentXProperty(), currentYProperty());}
+                        {
+                            bind(currentXProperty(),
+                                    currentYProperty(),
+                                    seriesLabel,
+                                    xAxisLabel,
+                                    yAxisLabel);
+                        }
                         @Override protected String computeValue() {
-                            String seriesName = series != null ? series.getName() : "";
-                            return seriesName + " X Axis is " + getCurrentX() + " Y Axis is " + getCurrentY();
+                            String seriesName = seriesLabel.getValue();
+                            String xAxisName = xAxisLabel.getValue();
+                            String yAxisName = yAxisLabel.getValue();
+                            String format = ControlResources.getString("XYChart.series.accessibleText");
+                            MessageFormat mf = new MessageFormat(format);
+                            Object[] args = {seriesName, xAxisName, getCurrentX(), yAxisName, getCurrentY()};
+                            String retVal = mf.format(args);
+                            System.out.println("result = " + retVal);
```

Comment:
You'll need to remove this debug print statement.

Reply:
> You'll need to remove this debug print statement.

Done.